internal/events/rabbitmq/publisher/publish.go

```diff
@@ -138,4 +131,7 @@ func (r *rabbitmqPublisher) publishMessage(msg message) {
 
 	log.Ctx(msg.Context).Info().Str("topic", msg.Topic).Msg("message published")
 	r.wg.Done()
+	r.lastPublishedAtMux.Lock()
+	r.lastPublishedAt = time.Now()
+	r.lastPublishedAtMux.Unlock()
```

Comment:
can we get a function to do this, since you've already extracted a bunch of methods and this is done in multiple places